src/map/lua/luautils.cpp

```diff
@@ -903,7 +903,16 @@ namespace luautils
         // clang-format off
         lua.set_function("GetFirstID", [&](std::string const& name) -> uint32
         {
-            return lookup[name].front();
+            if (lookup.find(name) != lookup.end())
+            {
+                return lookup[name].front();
+            }
+            else
+            {
+                ShowWarning(fmt::format("GetFirstID({}): Returning nil", name));
+
+                return 0;
```

Comment:
Nil here should be `sol::lua_nil` if the intent is to return a nil and not '0' value

Reply:
sorry, copied the debug print from the GetTableOfIDs, this returns zero as the function is expected to return a number. If we return nil it would still prevent a full crash, but break more things like `ID.mob.DEATH_JACKET + 5` (which may be a good thing)

Reply:
```
        // Update GetFirstID to use this new lookup
        // clang-format off
        lua.set_function("GetFirstID", [&](std::string const& name) -> std::optional<uint32>
        {
            if (lookup.find(name) != lookup.end())
            {
                return lookup[name].front();
            }
            else
            {
                ShowWarning(fmt::format("GetFirstID({}): Returning nil", name));

                return std::nullopt;
            }
        });
```

Untested, but this is what I would try

Reply:
I'm 95% sure this'll work, since all types in Lua are nullable. Or rather,  they're not complex enough to not be nullable. 

In case it doesn't work: since this is bound directly into Lua with `set_function`, you won't need to stuff the stuff into `std::optional<uint32>`. Instead, you can just use `sol::object`, so it'll be able to take both a number (maybe?) and `sol::lua_nil`

Reply:
This indeed worked:
![image](https://github.com/LandSandBoat/server/assets/131182600/60ed2188-8a05-4c90-aefb-c400667edee0)

Reply:
I would suggest using ShowError instead of warn for GetFirstID as well, since we'd want to catch that in CI on startup, even if doesn't fail later on

Reply:
By that logic shouldn't GetTableOfIDs also show error? I would think either of these functions being used and returning nothing should be the same cause for concern

Reply:
Did the thing:
![image](https://github.com/LandSandBoat/server/assets/131182600/88492b01-e38b-44ea-871f-c1d6b40eec1a)